cmd/client-s3.go

```diff
@@ -1416,6 +1416,8 @@ func (c *S3Client) Stat(ctx context.Context, opts StatOptions) (*ClientContent,
 		// so we can check if there is such prefix which exists
 		ctnt, err := c.getObjectStat(ctx, bucket, object, minio.StatObjectOptions{ServerSideEncryption: opts.sse, VersionID: opts.versionID})
 		if err == nil {
+			// HEAD with a version ID will not return version in the metadata
+			ctnt.VersionID = opts.versionID
```

Comment:
Maybe, instead we make c.getObjectStat() or even more lower level to return ctnt.VersionID if available

Reply:
that's a good idea, done.